challenge.rb

```diff
@@ -1,26 +1,32 @@
 
 def capitalize_each_string(input)
   #implement your solution here
+  input.collect {|a| a.capitalize}
 end
 
 def fetch_the_dog(input)
   #implement your solution here
+  input.select {|i| i == "dog"}
 end
 
 def no_dogs_allowed(input)
   #implement your solution here
+  input.reject {|i| i == "dog"}
 end
 
 def count_the_animals(input)
   #implement your solution here
+  input.count
 end
 
 def fetch_the_first_two(input)
   #implement your solution here
+  input.first(2)
 end
 
 def fetch_CD_animals(input)
   #implement your solution here
+  input.find_all{|i| i[0] =~ /[cd]/ }
```

Comment:
Also, it turns out I didn't really understand why that line was working.  It'd be conceptually simpler to use something like this in the block: 

> /^[cd]/ === i